### ISSUE TITLE
Astro component parser breaks with JSX logical operator when multiple components are returned

### ISSUE THREAD

**Issue by jhuleatt** [withastro/astro#10170]
### Astro Info

```block
Astro                    v4.4.1
Node                     v20.10.0
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/sitemap
                         @astrojs/preact
                         @astrojs/react
                         @astrojs/mdx
```

### Describe the Bug

I'm conditionally showing some HTML in an Astro component as documented [here](https://docs.astro.build/en/basics/astro-syntax/#dynamic-html).

However, when my expression returns multiple HTML elements, I get an error like `Expected "}" but found "data"`. Here's an example expression that causes that error:

```jsx
{true && <span>&rarr;</span><span>hello</span>}
```

I know it's valid syntax because if I break it into multiple lines, it works as expected:

```jsx
{true && 
	<span>&rarr;</span>
	<span>hello</span>
}
```

Also, if the expression only returns a single HTML element, that works fine too:

```jsx
{true && <span>&rarr;</span>}
```

It only breaks when the expression is on a single line and returns multiple HTML elements, as shown in the first code snippet.

### What's the expected result?

This statement renders without error:

```jsx
{true && <span>&rarr;</span><span>hello</span>}
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-d8xxzx?file=src%2Fcomponents%2FCard.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
If you wrap it with parens does it fix it? `{true && 
	(<span>&rarr;</span>
	<span>hello</span>)
}`?

**Comment by matthewp** [comment]
Yeah, looks like it does. We should probably just document that this needs to be wrapped and perhaps error (if possible) in the compiler if you don't do so. I'll see if anyone wants to work on this.
